server/auto_updater/dataset_updater.py

```diff
@@ -0,0 +1,292 @@
+import csv
+from collections import defaultdict, Counter
+import lzma
+from pathlib import Path
+import pickle
+import tarfile
+import urllib.request as request
+
+import lxml.etree as ET
+import numpy as np
+import pandas as pd
+import spacy
+import tqdm
+
+nlp = spacy.load("en_core_web_sm")
```

Comment:
@danich1: I forgot to tell you: if you want to run `generate_new_papers()` function on the locally downloaded `comm_use.A-B.xml.tar.gz`, you need to add the following line right after line 15:
```nlp.max_length = 1600000```
otherwise you would get the following error after about 2.5 hours:
```
Traceback (most recent call last):
  File "./1_generate_new_papers.py", line 174, in <module>
    dhu_generate_new_papers(
  File "./1_generate_new_papers.py", line 90, in dhu_generate_new_papers
    doc_vector, word_counter = generate_vector_counts(
  File "./1_generate_new_papers.py", line 151, in generate_vector_counts
    nlp(all_text),
  File "/home/dhu/.venv/pss/lib/python3.8/site-packages/spacy/language.py", line 977, in __call__
    doc = self.make_doc(text)
  File "/home/dhu/.venv/pss/lib/python3.8/site-packages/spacy/language.py", line 1056, in make_doc
    raise ValueError(
ValueError: [E088] Text of length 1347496 exceeds maximum of 1000000. The parser and NER models
require roughly 1GB of temporary memory per 100,000 characters in the input. This means long
texts may cause memory allocation errors. If you're not using the parser or NER, it's probably
safe to increase the `nlp.max_length` limit. The limit is in number of characters, so you can
check whether your inputs are too long by checking `len(text)`.
```
I suspect that setting `nlp.max_length` to any number larger than 1347496 may work, but I used `1600000` to err on the side of caution.